File: leetcode/217/217a.cpp
const int ZERO = [](){
    ios::sync_with_stdio(false);
    cin.tie(nullptr);
    return 1;
}();

class Solution {
public:
    bool containsDuplicate(vector<int>& nums) {
        unordered_set<int> nums_set;
        for(int e: nums){
            if (!nums_set.insert(e).second) {
                return true;
            }
        }
        return false;
    }
};

File: leetcode/217/217a.rs
use std::collections::HashSet;

impl Solution {
    pub fn contains_duplicate(nums: Vec<i32>) -> bool {
        if nums.len() < 2 {
            return false;
        }
        let mut map = HashSet::new();
        nums.iter().any(|&v| !map.insert(v))
    }
}